packages/openapi3/test/xml-models.test.ts

```diff
@@ -0,0 +1,756 @@
+  it("provide the namespace and prefix", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @ns("https://example.com/ns1", "ns1")
+        model Book {
+          id: string;
+        };`,
+    );
+    expect(res.schemas.Book).toMatchObject({
+      type: "object",
+      properties: {
+        id: {
+          type: "string",
+        },
+      },
+      required: ["id"],
+      xml: {
+        namespace: "https://example.com/ns1",
+        prefix: "ns1",
+      },
+    });
+  });
+
+  it("provide the namespace and prefix using string", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `model Book {
+          @ns("https://example.com/ns1", "ns1")
+          id: string;
+        };`,
+    );
+    expect(res.schemas.Book).toMatchObject({
+      type: "object",
+      properties: {
+        id: {
+          type: "string",
+          xml: {
+            namespace: "https://example.com/ns1",
+            prefix: "ns1",
+          },
+        },
+      },
+      required: ["id"],
+    });
+  });
+
+  it("provide the namespace and prefix using enum on model", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @nsDeclarations
+        enum Namespaces {
+          smp:"http://example.com/schema",
+        }
+
+        @ns(Namespaces.smp)
+        model Book {
+          id: string;
+        };`,
+    );
+    expect(res.schemas.Book).toMatchObject({
+      type: "object",
+      properties: {
+        id: {
+          type: "string",
+        },
+      },
+      required: ["id"],
+      xml: {
+        namespace: "http://example.com/schema",
+        prefix: "smp",
+      },
+    });
+  });
+
+  it("provide the namespace and prefix using enum on model and properties", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @nsDeclarations
+        enum Namespaces {
+          smp:"http://example.com/schema",
+          ns2:"http://example.com/ns2",
+        }
+
+        @ns(Namespaces.smp)
+        model Book {          
+          id: string;
+          @ns(Namespaces.smp)
+          title: string;
+          @ns(Namespaces.ns2)
+          author: string;
+        };`,
+    );
+    expect(res.schemas.Book).toMatchObject({
+      type: "object",
+      properties: {
+        id: {
+          type: "string",
+        },
+        title: {
+          type: "string",
+          xml: {
+            namespace: "http://example.com/schema",
+            prefix: "smp",
+          },
+        },
+        author: {
+          type: "string",
+          xml: {
+            namespace: "http://example.com/ns2",
+            prefix: "ns2",
+          },
+        },
+      },
+      required: ["id", "title", "author"],
+      xml: {
+        namespace: "http://example.com/schema",
+        prefix: "smp",
+      },
+    });
+  });
+});
+describe("Array of primitive types", () => {
+  it("unwrapped tags array in the xmlBook model.", async () => {
```

Comment:
I think the unwrapped cases with array are broken

[Playgroun example](https://cadlplayground.z22.web.core.windows.net/prs/4214/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZeG1sIjsKCnVzaW5nIFhtbDvHC0h0dHA7CgpAc2VydmljZSh7CiAgdGl0bGU6ICJXaWRnZXQgU8YbIiwKfSkKbmFtZXNwYWNlIERlbW%2FHGzsKCm9wIHRlc3QoQGhlYWRlciBjb250ZW50VHlwxEphcHBsaWNhdGlvbuUAiiwgQGJvZHkgYm9vazogQm9vayk6IHZvaWQ7Cgptb2RlbMUUIOQAjcRxOiBzdHJpbmc7CgogIEBlbmNvZGVkTmFtZSjTWiJYbWzEHCIpCiAgcmXEQGTNQWxpc3RPZlPFEXPIGltd313HXcw9WG1sxWXNUVLOcsdYdGHfa9NrxDTIY8QM1lpAdW53cmFwcGVkCiDHGlXIEs0r0CzEU8sqVGHWJ%2F8Ap%2BYAp8lI7ACwyRXpALnHZH3oAeVUYWfzAeR9Cg%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40typespec%2Fhttp%2Fall%22%5D%7D%7D) 

the unwrapped node should still use the property name not the target type name. That is only if its wrapped